qa/tasks/cephfs/test_client_recovery.py

```diff
@@ -871,3 +871,49 @@ def test_client_eviction_if_config_is_unset(self):
             self.mount_a.mount_wait()
             self.mount_a.create_destroy()
             self.clear_laggy_params(osd)
+
+    def test_mounting_clients_after_killed(self):
+        """
+        If a client is killed, an OSD is laggy and config option
+        defer_client_eviction_on_laggy_osds is set then test spawning
+        new clients works fine.
+        """
+        self.config_set('mds', 'mds_defer_session_stale', 'true')
+        self.assertEqual(self.config_get('mds', 'mds_defer_session_stale'),
+                         'true')
+        self.config_set('mds', 'defer_client_eviction_on_laggy_osds', 'true')
+        self.assertEqual(self.config_get(
+            'mds', 'defer_client_eviction_on_laggy_osds'), 'true')
+
+        # make an OSD laggy
+        osd = self.get_a_random_osd()
+        self.make_osd_laggy(osd)
+
+        mount_a_gid = self.mount_a.get_global_id()
+        self.mount_a.kill()
+        self.assert_session_state(mount_a_gid, "open")
+
+        time.sleep(self.fs.get_var("session_timeout") * 1.5)
+
+        # it takes time to have laggy clients entries in cluster log,
+        # wait for 6 minutes to see if it is visible, finally restart
+        # the clients
+        with contextutil.safe_while(sleep=5, tries=6) as proceed:
+            while proceed():
+                try:
+                    with self.assert_cluster_log("1 client(s) laggy due to"
+                                                 " laggy OSDs",
+                                                 timeout=55):
+                        # make sure clients weren't evicted
+                        self.assert_session_count(2)
+                        break
+                except (AssertionError, CommandFailedError) as e:
+                    log.debug(f'{e}, retrying')
+
+        self.mount_a.umount_wait()
```

Comment:
With a suspended netns, do you expect this succeed? (I sure hope it doesn't...)

Reply:
might be, i was hoping that the client should get blocklisted and this should take care of it:
https://github.com/ceph/ceph/blob/7d1f6d70f19b0144e31140a7887f5e2cd14d2dd0/qa/tasks/cephfs/fuse_mount.py#L321

plus the fact that it reaches `self.mount_a.create_destroy()` should mean the umount did went fine, right https://github.com/ceph/ceph/pull/55784#issuecomment-1968770427

Reply:
That won't run because the client isn't blocklisted. I would expect this `umount_wait` to suspend until the MDS finally evicts the client after ~300 seconds.

In which case, I'm not sure what the point of this test is.